docs/running-clusterfuzzlite/gitlab.md

```diff
@@ -0,0 +1,229 @@
+---
+layout: default
+parent: Step 2&#58; Running ClusterFuzzLite
+grand_parent: ClusterFuzzLite
+title: GitLab
+nav_order: 1
+permalink: /running-clusterfuzzlite/gitlab/
+---
+# GitLab
+{: .no_toc}
+
+- TOC
+{:toc}
+---
+
+This page explains how to set up ClusterFuzzLite to run on [GitLab].
+To get the most of this page, you should have already set up your
+[build integration] and read the more
+[high-level document on running ClusterFuzzLite].
+
+## Gitlab runner
+The following examples use a `docker` gitlab runner running sibling containers:
+[doc](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-socket-binding).
+
+It should be possible to achieve the same functionalities with using a shell executor.
+But then, the `.gitlab-ci.yml` should be different, and explicitly call the `docker` commands
+on clusterfuzz-lite images.
+
+Docker-in-Docker does not seem possible as clusterfuzz-lite images
+do not have the docker tools installed.
+
+## .gitlab-ci.yml
+For basic ClusterFuzzLite functionality, all you need is a single job
+to enable fuzzing on your merge requests.
+
+To enable more features, we recommend having different jobs for:
+
+- continuous builds
+- batch fuzzing and corpus pruning
+- coverage
+
+### MR fuzzing
+
+To add a fuzzing job that fuzzes all merge requests to your repo, add the
+following default configurations to `.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+variables:
+  SANITIZER: "address"
+  CFL_PLATFORM: gitlab
+
+clusterfuzzlite:
+  image:
+    name: google/clusterfuzzlite/actions/build_fuzzers@v1
+    entrypoint: [""]
+  stage: fuzz
+  rules:
+    # default code change
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      variables:
+        MODE: "code-change"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    # will build and run the fuzzers
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    # upload artifacts when a crash makes the job fail
+    when: always
+    paths:
+      - artifacts/crashes/
+```
+{% endraw %}
+
+You may also wish to set [tags](https://docs.gitlab.com/runner/#tags) to select a relevant runner.
+
+Optionally, edit the following variables to customize your settings:
+- `SANITIZER` Change or enable more sanitizers.
+- `LANGUAGE` Define the language of your project.
+- `CFL_BRANCH` Branch to fuzz, default is `CI_DEFAULT_BRANCH`.
+- `FILESTORE` To use corpus produced by other jobs.
+- `FUZZ_SECONDS` Change the amount of time spent fuzzing.
+- `CFL_ARTIFACTS_DIR` : To save your artifacts in a different directory than `artifacts`
+
+For `SANITIZER`, you may also use a matrix to use multiple sanitizers with the same job.
+
+### Batch fuzzing and corpus pruning
```

Comment:
Because both functions update the corpus.

In Gitlab, there is only one gitlab-ci.yml (which has different jobs), and the schedules are defined independently.
The way to know which schedule triggered is to use some variables.
The job is the same for these 2 functions.

Reply:
I can see your point from a technical view on the jobs being identical. But from a higher-level point of view, these are different things. So I still would like to see two jobs, one for batch fuzzing (these could be three already with all three sanitizers), and one for pruning. Using the same name for the four jobs is misleading IMHO.